src/Controller/HearingController.php

```diff
@@ -417,11 +424,59 @@ public function hearingPostApprove(CaseEntity $case, HearingPostResponse $hearin
             $this->entityManager->persist($document);
 
             $digitalPostHelper->createDigitalPost($document, $documentTitle, get_class($case), $case->getId(), [], $digitalPostRecipients);
+
+            // Create digital post case event (sagshændelse)
+            $digitalPostCaseEvent = new CaseEvent();
+            $digitalPostCaseEvent->setCaseEntity($case);
+            $digitalPostCaseEvent->setCategory(CaseEvent::CATEGORY_OUTGOING);
+
+            $caseEventPartyRelation = new CaseEventPartyRelation();
+            $caseEventPartyRelation->setParty($sender);
+            $caseEventPartyRelation->setCaseEvent($digitalPostCaseEvent);
+            $caseEventPartyRelation->setType(CaseEventPartyRelation::TYPE_RECIPIENT);
+            $this->entityManager->persist($caseEventPartyRelation);
+
+            $digitalPostCaseEvent->setSubject(CaseEvent::SUBJECT_HEARING_CONTRADICTIONS_BRIEFING);
+            $digitalPostCaseEvent->setReceivedAt(new DateTime('now'));
+            $digitalPostCaseEvent->setCreatedBy($caseEventUser);
+
+            $digitalPost = $digitalPostRepository->findByDocumentAndCase($document, $case);
+
+            if (1 !== count($digitalPost)) {
+                throw new CaseEventException(sprintf('Should find one DigitalPost but found %d', count($digitalPost)));
+            }
+
+            $digitalPostCaseEvent->setDigitalPost(reset($digitalPost));
+            $this->entityManager->persist($digitalPostCaseEvent);
+        }
+
+        // Create case event (sagshændelse)
+        $caseEvent = new CaseEvent();
+        $caseEvent->setCaseEntity($case);
+        $caseEvent->setCategory(CaseEvent::CATEGORY_INCOMING);
+
+        $caseEventPartyRelation = new CaseEventPartyRelation();
+        $caseEventPartyRelation->setParty($hearingPost->getSender());
+        $caseEventPartyRelation->setCaseEvent($caseEvent);
+        $caseEventPartyRelation->setType(CaseEventPartyRelation::TYPE_SENDER);
+        $this->entityManager->persist($caseEventPartyRelation);
+
+        $caseEvent->setSubject(CaseEvent::SUBJECT_HEARING_CONTRADICTIONS_BRIEFING);
+        $caseEvent->setReceivedAt(new DateTime('now'));
+        $caseEvent->setCreatedBy($caseEventUser);
+
+        $caseEvent->addDocument($hearingPost->getDocument());
+
+        foreach ($hearingPost->getAttachments() as $attachment) {
+            $caseEvent->addDocument($attachment->getDocument());
```

Comment:
Can these setter calls (and the adder call) be moved up to the other setter/adder calls on `$caseEvent`?

Reply:
Moved to `CaseEventHelper`.